tests/backup/namespace_labelled_backup_test.go

```diff
@@ -0,0 +1,197 @@
+					defer wg.Done()
+					err := CreateBackupWithNamespaceLabel(backupName, SourceClusterName, bkpLocationName, backupLocationUID,
+						labelSelectors, orgID, srcClusterUid, "", "", "", "", mapToString(labels), listOfLabelledNamespaces, ctx)
+					log.FailOnError(err, "Failed while trying to take namespace labelled backup of applications %v", listOfLabelledNamespaces)
+				}(backupName)
+			}
+			wg.Wait()
+			log.InfoD("List of namespace labelled backups- %v", backupNames)
+		})
+		Step("Share backup with users with different access level", func() {
+			log.InfoD("Share backup with users with different access level")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			accessUserBackupContext, err = ShareBackupWithUsersAndAccessAssignment(backupNames, users, ctx)
+			log.FailOnError(err, fmt.Sprintf("Sharing backup %s with users %v", backupNames, users))
+		})
+		Step("Validate if the users with different access level can restore/delete backup as per their access mode", func() {
+			log.InfoD("Validate if the users with different access level can restore/delete backup as per their access mode")
+			for key, val := range accessUserBackupContext {
+				restoreName := fmt.Sprintf("%s-%s-%v", key.user, RestoreNamePrefix, time.Now().Unix())
+				access := key.accesses
+				if access != ViewOnlyAccess {
+					log.Infof("Adding the restores %v which will be created while validating %v access", restoreName, backupAccessKeyValue[access])
+					userRestoreContext[key.context] = restoreName
+				}
+				ValidateSharedBackupWithUsers(key.user, key.accesses, val, restoreName)
+			}
+		})
+	})
+	JustAfterEach(func() {
+		var wg sync.WaitGroup
+		defer EndPxBackupTorpedoTest(contexts)
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = true
+		ValidateAndDestroy(contexts, opts)
+		log.Infof("Generating user context")
+		for _, userName := range users {
+			ctxNonAdmin, err := backup.GetNonAdminCtx(userName, commonPassword)
+			log.FailOnError(err, "Fetching non admin ctx")
+			userContextsList = append(userContextsList, ctxNonAdmin)
+		}
+		log.Infof("Deleting restore created by users")
+		for userContext, restoreName := range userRestoreContext {
+			err = DeleteRestore(restoreName, orgID, userContext)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting Restore %s", restoreName))
+		}
+		log.Infof("Deleting registered clusters for non-admin context")
+		for _, ctxNonAdmin := range userContextsList {
+			CleanupCloudSettingsAndClusters(make(map[string]string), "", "", ctxNonAdmin)
+		}
+		log.Infof("Cleaning up users")
+		for _, userName := range users {
+			wg.Add(1)
+			go func(userName string) {
+				defer wg.Done()
+				err := backup.DeleteUser(userName)
+				dash.VerifySafely(err, nil, fmt.Sprintf("Deleting user %v", userName))
+			}(userName)
+		}
+		wg.Wait()
```

Comment:
for this also create a ticket for cleanup of users/restores.
and deleting restore namespaces.

Reply:
https://portworx.atlassian.net/browse/PA-815

deleting restore namespaces: Tarun is working on it